R/sf.R

```diff
@@ -48,14 +50,23 @@ as_points <- function(df, xvar = "x", yvar = "y", remove = FALSE) {
     assert_that(has_name(df, xvar))
     assert_that(has_name(df, yvar))
 
+    x_var <- parse_expr( xvar )
+    y_var <- parse_expr( yvar )
     df_cleaned <-
-        df[!is.na(df[,xvar]) & !is.na(df[,yvar]),]
+        df %>%
+        filter(!is.na(!!xvar) & !is.na(!!yvar))
 
     if (nrow(df_cleaned) < nrow(df)) {
         warning(nrow(df) - nrow(df_cleaned),
                 " locations were removed because of missing X or Y coordinates.")
     }
 
+    check_dubbels <- df_cleaned %>% distinct(!!xvar, !!yvar)
+    if (nrow(check_dubbels) < nrow(df)) {
+        warning(nrow(df) - nrow(check_dubbels),
+                " locations are duplicated.")
```

Comment:
I think it may be meaningful to draw the user's attention to duplicated coordinates, depending on the case at hand: after all the function is more generic than its use in Watina only. What about this wording:

```suggestion
                " rows share common coordinates.")
```